tests/functional_test_base.go

```diff
@@ -239,12 +239,26 @@ func GetTestClusterConfig(configFile string) (*TestClusterConfig, error) {
 	// #nosec
 	confContent, err := os.ReadFile(configLocation)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read test cluster config file %v: %v", configLocation, err)
+		return nil, fmt.Errorf("failed to read test cluster config file %s: %w", configLocation, err)
 	}
 	confContent = []byte(os.ExpandEnv(string(confContent)))
 	var options TestClusterConfig
 	if err := yaml.Unmarshal(confContent, &options); err != nil {
-		return nil, fmt.Errorf("failed to decode test cluster config %v", err)
+		return nil, fmt.Errorf("failed to decode test cluster config %s: %w", configLocation, err)
+	}
+
+	// If -FaultInjectionConfigFile is passed to the test runner,
+	// then fault injection config will be added to the test cluster config.
+	if TestFlags.FaultInjectionConfigFile != "" {
+		fiConfigContent, err := os.ReadFile(TestFlags.FaultInjectionConfigFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read test cluster fault injection config file %s: %v", TestFlags.FaultInjectionConfigFile, err)
+		}
+
+		// This will merge fault injection config content into the test cluster config.
```

Comment:
should you merge just into the `FaultInjection` field so it can't just overwrite other stuff?

Reply:
I was thinking about converting this to a set of aux config files.